### PR TITLE
BIM: Specific IFC icons

### LINKED ISSUE #14427: BIM: Native IFC projects use the same "IFC" icon for all objects
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

When creating a NativeIFC project, all items in the tree lose their original icons (building, wall, etc) and are replaced by the same "IFC package" icon. Even with small projects, that makes it visually difficult to navigate through elements while creating a model. There might be a specific motivation, but even to someone who has been using the BIM/Arch workbenches for a while, it feels like a regression/feature loss.

Before:
Captura de pantalla de 2024-06-01 14-45-42

After:
Captura de pantalla de 2024-06-01 14-46-44

The "After" capture was made with the attached file: open the file and manually drag and drop all tree entities to be under `_IfcProject`.

### Full version info

```shell
OS: Ubuntu Core 22 (ubuntu:GNOME/ubuntu)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.37573 (Git) Snap 1037
Build type: Release
Branch: main
Hash: 91aac05021023c946dc1545f2b13512ccf326550
Python 3.10.12, Qt 5.15.10, Coin 4.0.0, Vtk 7.1.1, OCC 7.7.1
Installed mods: 
  * OpenTheme 2024.5.16
  * OpenTheme.backup1716114997.8759558 2024.5.16 (Disabled)
  * Curves 0.6.36
  * ezydxf
  * ExplodedAssembly
  * BillOfMaterials 0.0.8.2
  * Ondsel-Lens 2024.5.6.01
  * Assembly4 0.50.12
```

### Subproject(s) affected?

BIM/Arch

### Anything else?

- [File to reproduce](https://github.com/user-attachments/files/15521829/IFC_icons.zip)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by yorikvanhavre**
NativeIFC objects will now look for an icon matching their class before using the default one.

Fixes #14427

This should be merged after 1.0

### CODE REVIEW

**Comment by maxwxyz** [comment]
why does this has to be post 1.0?

**Comment by yorikvanhavre** [comment]
Because it's a feature and we're in feature freeze...

File: src/Mod/BIM/Resources/icons/IFC/IfcBuildingStorey.svg
Comment (maxwxyz) [inline src/Mod/BIM/Resources/icons/IFC/IfcBuildingStorey.svg:1]: @marcuspollio didn't you make a new one for this?

Comment (marcuspollio) [reply src/Mod/BIM/Resources/icons/IFC/IfcBuildingStorey.svg:1]: Hi @maxwxyz
Not that's I aware of.
The "similar looking" one in PR 15079 is for Techdraw-ArchView.
This one is to organise the model objects into Storeys following the IFC schema.
Their use and concept are quite different. They could be made more distinctive though. For a latter time anyways.

**Pull request merged**